### PR TITLE
Record typing test

### PR DESCRIPTION
**Pull request by matthewhammer**
Reproduces a bug in the lowering or other front-end phases of the compiler, where the initial type-check passes, but post lowering, the IR does not carry enough type information to check anymore (my best guess).

Thanks to @ninegua for reporting to the team Slack today, based on another dev's comment.

I'm not committing the new test outcome, because it's erroneous.

This message appears:
```
Ill-typed intermediate code after Desugaring (use -v to see dumped IR):
(unknown location): IR type error [M0000], subtype violation:
  {var myList : List<None>}
  {var myList : List<Nat>}
```

### CODE REVIEW

File: test/run/records.mo [hunk 1]
```diff
@@ -23,3 +23,7 @@ assert(rec3.y == y);
 assert(rec3.z == 13);
 assert(rec3.a == 4);
 assert(rec3.b == 5);
+
+type List<X> = ?(X, List<X>);
+func nil<X>() : List<X> = null;
+let rec4 : { var myList : List<Nat> } = { var myList = nil() };
```
Comment (nomeata) [inline test/run/records.mo:29]: Probably better putting that regression tests into its own file (`issue3265.mo` maybe).

**Comment by crusso** [comment]
Investigating...

File: test/run/records.mo (hunk 1 above)
Comment (crusso) [reply test/run/records.mo:29]: Done as part of #3268

**Comment by crusso** [comment]
(Closing - fixed in #3268)

**Pull request closed without merging**